File: backend/src/Shalom.Application/Contracts/ReadingDayDto.cs
namespace Shalom.Application.Contracts;

/// <summary>One day of a reading plan. <paramref name="CompletedOn"/> is the local day it was read, or null.</summary>
public record ReadingDayDto(Guid Id, int DayNumber, string PassageReference, string YouVersionUrl, DateOnly? CompletedOn);

File: frontend/projects/api/src/lib/models/reading-day-dto.ts
/** One day of a reading plan. `completedOn` is the local day it was read, or null. */
export interface ReadingDayDto {
  readonly id: string;
  readonly dayNumber: number;
  readonly passageReference: string;
  readonly youVersionUrl: string;
  readonly completedOn: string | null;
}
